lib/new-settings-loader.js

```diff
@@ -40,6 +53,18 @@ function loadAccounts (schemaVersion) {
     .then(_.compose(_.defaultTo({}), _.get('data.accounts')))
 }
 
+function showAccounts (schemaVersion) {
+  return loadAccounts(schemaVersion)
+    .then(accounts => {
+      const filledSecretPaths = _.compact(_.map(path => {
+        if (!_.isEmpty(_.get(path, accounts))) {
+          return path
+        }
+      }, SECRET_FIELDS))
+      return _.compose(_.map(path => _.assoc(path, PASSWORD_FILLED), filledSecretPaths))(accounts)
+    })
+}
+
```

Comment:
This looks like a good use case for lenses. Ramda has them but for lodash maybe we could use something like:

https://github.com/jamesmcnamara/shades

Reply:
Hey, Josh, I've been going through the docs, and I don't seem to find a way to make this return much simpler. Do you have any suggestions? It would be nice if shades had a way to replace all the filledSecretPaths without a map, right?

Reply:
Hey José, the idea of lenses is you can define a path down into the structure and then read and set that deep field easily. I didn't look into how exactly it would change the code here, though. I'll merge this for now but you can play around with it afterwards.